components/crypto/src/proto/schema/types.proto

```diff
@@ -16,6 +16,8 @@ syntax = "proto3";
 
 package exonum.crypto;
 
+option java_package = "com.exonum.binding.messages";
```

Comment:
I don't really like idea of making framework know something about client code.

Reply:
It is not the client code, it is (generated) framework code.

Reply:
IMHO from side of framework everything that depends on it is the client code. We can rephrase it like "I don't really like idea of making base entity know something about child entity"

Reply:
> making base entity know something about child entity

In fact, the base entity does not lean on this piece of information, it's kinda annotation for the protobuf compiler _in case_ entities are _built for Java_.
I believe there are only two real options for configuring this for Java: specifying it in a `.proto` file via `option` or patching it (looks like there is no way specifying it via option for cmd line `protoc` or maven plugin). Personally I like more the first approach - you only need to do this once per platform (and this does not affect others).

Reply:
The need for such directives indicates on design flaws in protobuf.